profiles/common/modules/features/nexteuropa_editorial/includes/nexteuropa_editorial.tracking.inc

```diff
@@ -0,0 +1,429 @@
+<?php
+/**
+ * @file
+ * Code specific to customization for the change tracking management.
+ */
+
+/**
+ * Field widget form alter callback.
+ *
+ * It inject a specific validate callback in achrge of scanning field value.
+ *
+ * @see nexteuropa_editorial_field_widget_text_textarea_form_alter()
+ * @see nexteuropa_editorial_field_widget_text_textarea_with_summary_form_alter()
+ * @see nexteuropa_editorial_field_widget_text_textfield_form_alter()
+ */
+function _nexteuropa_editorial_field_widget_tracking_form_alter(&$element, &$form_state, $context) {
+  // So far, only nodes are supported.
+  $entity = FALSE;
+  if (isset($element['#entity'])) {
+    $entity = $element['#entity'];
+    // Workaround for entities that do not use "entity_load" correctly
+    // like "comment".
+    if (!isset($entity->entity_type) && isset($element['#entity_type'])) {
+      $entity->entity_type = $element['#entity_type'];
+    }
+  }
+  $tracking_settings = _nexteuropa_editorial_wysiwyg_tracking_settings($entity);
+  if ($tracking_settings['is_activate']) {
+    $instance_settings = $context['instance']['settings'];
+    if (isset($instance_settings['text_processing']) && $instance_settings['text_processing']) {
+      // Tracking change is only possible on fields on which
+      // a text format can be set.
+      $element['#element_validate'][] = '_nexteuropa_editorial_tracking_field_validate';
+    }
+  }
+}
+
+/**
+ * Flags the entity if it or its translations contain tracked changes.
+ *
+ * The check on translations is only done if entity translations are
+ * managed by "Entity translation".
+ *
+ * @param object $entity
+ *   The entity to treat.
+ * @param string $type
+ *   The entity type.
+ *
+ * @throws EntityMalformedException
+ *   If the entity bundle is not set.
+ */
+function _nexteuropa_editorial_set_tracking_flags($entity, $type = 'node') {
+  // Tracking changes are only present in these "fields type".
+  $screened_field_types = array(
+    'text_textarea',
+    'text_textarea_with_summary',
+    'text_textfield',
+  );
+  list(,, $bundle) = entity_extract_ids($type, $entity);
+  $entity_language = entity_language($type, $entity);
+
+  if (!isset($entity->tracking_activate)) {
+    $entity->tracking_activate = array();
+  }
+
+  // Content translation case.
+  $available_transl_langs = array($entity_language);
+  if (isset($entity->translations)) {
+    // Entity translation case.
+    $available_transl = $entity->translations;
+    $available_transl_langs = array_keys($available_transl->data);
+  }
+
+  $ent_field_instances = field_info_instances($type, $bundle);
+  // Note: special field types like field collection or paragraphs
+  // are not supported yet.
+  foreach ($ent_field_instances as $field_name => $ent_field_instance) {
```

Comment:
I don't think we should limit this to one field, so i would not follow this remark. Or maybe I didn't understand?